packages/typespec-ts/src/index.ts

```diff
@@ -195,6 +196,7 @@ export async function $onEmit(context: EmitContext) {
         buildModelsOptions(modularCodeModel, subClient);
         const hasClientUnexpectedHelper =
           needUnexpectedHelper.get(subClient.rlcClientName) ?? false;
+        buildSerializers(modularCodeModel);
```

Comment:
I remember in `serializeUtil.ts` we also generate serialize function for each union variant, just curious if it's possible to leverage the `buildSerializers` logic instead of generating it there?